src/app/components/pages/CommunitiesIndex.jsx

```diff
@@ -19,20 +20,25 @@ export default class CommunitiesIndex extends React.Component {
             communities_idx,
             username,
             walletUrl,
+            loading,
         } = this.props;
         const ordered = communities_idx.map(name => communities.get(name));
 
-        if (communities_idx.length == 0) {
+        if (communities_idx.size === 0) {
             return (
                 <center>
-                    <h5>Loading...</h5>
+                    <LoadingIndicator
+                        style={{ marginBottom: '2rem' }}
+                        type="circle"
+                    />
+                    Loading...
```

Comment:
Probably don't need to show the text 'Loading...' if the spinner is being shown.